go/vt/vtgate/scatter_conn_test.go

```diff
@@ -615,3 +496,10 @@ func TestShuffleQueryParts(t *testing.T) {
 	// Return the generator to what it was to avoid disrupting other tests.
 	injectShuffleQueryPartsRandomGenerator(oldGen)
 }
+
+func newTestScatterConn(hc discovery.HealthCheck, topoServer topo.Server, serv topo.SrvTopoServer, statsName, cell string, retryCount int, tabletTypesToWait []topodatapb.TabletType) *ScatterConn {
```

Comment:
tabletTypesToWait is always nil when calling this (please check though), I think you can remove the parameter entirely, and not even call gateway.WaitForTablets.